x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/inference/Rerank.java

```diff
@@ -166,4 +172,19 @@ public LogicalPlan surrogate() {
         Order sortOrder = new Order(source(), scoreAttribute, Order.OrderDirection.DESC, Order.NullsPosition.ANY);
         return new OrderBy(source(), this, List.of(sortOrder));
     }
+
+    @Override
+    public List<Attribute> output() {
+        if (lazyOutput == null) {
+            lazyOutput = planHasAttribute(child(), scoreAttribute)
+                ? child().output()
+                : mergeOutputAttributes(List.of(scoreAttribute), child().output());
+        }
+
+        return lazyOutput;
+    }
+
+    public static boolean planHasAttribute(QueryPlan<?> plan, Attribute attribute) {
```

Comment:
Maybe this could be included in QueryPlan instead, as it's used outside this class?

Reply:
The check that is done (through an `equalTo` can be misleading so I would prefer to keep it here for the moment.